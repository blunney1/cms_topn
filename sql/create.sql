--
--Testing cms_topn function of the extension which creates new cms_topn structure
--

CREATE EXTENSION cms_topn;

--check errors for unproper parameters
SELECT cms_topn(0);
SELECT cms_topn(1, -0.1, 0.9);
SELECT cms_topn(-1, 0.1, 0.9);
SELECT cms_topn(3, 0.1, -0.5);
SELECT cms_topn(4, 0.02, 1.1);

--check proper parameters
CREATE TABLE create_test (
	cms_topn_column cms_topn
);

INSERT INTO create_test values(cms_topn(10));
INSERT INTO create_test values(cms_topn(5, 0.01, 0.95));
SELECT cms_topn_info(cms_topn_column) from create_test;
