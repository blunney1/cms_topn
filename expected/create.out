--
--Testing cms_topn function of the extension which creates new cms_topn structure
--
CREATE EXTENSION cms_topn;
--check errors for unproper parameters
SELECT cms_topn(0);
ERROR:  invalid parameters for cms_topn
HINT:  Number of top items has to be positive
SELECT cms_topn(1, -0.1, 0.9);
ERROR:  invalid parameters for cms_topn
HINT:  Error bound has to be between 0 and 1
SELECT cms_topn(-1, 0.1, 0.9);
ERROR:  invalid parameters for cms_topn
HINT:  Number of top items has to be positive
SELECT cms_topn(3, 0.1, -0.5);
ERROR:  invalid parameters for cms_topn
HINT:  Confidence interval has to be between 0 and 1
SELECT cms_topn(4, 0.02, 1.1);
ERROR:  invalid parameters for cms_topn
HINT:  Confidence interval has to be between 0 and 1
--check proper parameters
CREATE TABLE create_test (
	cms_topn_column cms_topn
);
INSERT INTO create_test values(cms_topn(10));
INSERT INTO create_test values(cms_topn(5, 0.01, 0.95));
SELECT cms_topn_info(cms_topn_column) from create_test;
                   cms_topn_info                    
----------------------------------------------------
 Sketch depth = 6, Sketch width = 1360, Size = 63kB
 Sketch depth = 3, Sketch width = 272, Size = 6kB
(2 rows)

